orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db_compressed.sh

```diff
@@ -0,0 +1,254 @@
+
+  # print to the terminal if we have one
+  test -t 1 && echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg"
+  echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg" >>$LOGFILE
+}
+
+usage() {
+  doLog "There might be missing values in /srv/pillar/postgresql/disaster_recovery.sls or /srv/pillar/postgresql/postgre.sls."
+  doLog "This might be caused by the command not being run on the Primary Gateway node or due to never having run a backup/restore via the CDP CLI before."
+  doLog "Script accepts the following inputs:"
+  doLog "  -s path : Object Storage Service url to place backups."
+  doLog "  -h host : PostgresSQL host name."
+  doLog "  -p port : PostgresSQL port."
+  doLog "  -u username : PostgresSQL user name."
+  doLog "  -r group : Ranger admin group."
+  doLog "  -c bool : Whether or not to close connections for the database while it is being backed up."
+  doLog "  -z compression_level : (optional)  database dump compression level 0-9"
+  doLog "  -d \"db1 db2 db3\" : (optional) Names of the databases to backup. If not given, will backup ranger and hive databases."
+}
+
+BACKUP_LOCATION=""
+HOST=""
+PORT=""
+USERNAME=""
+RANGERGROUP=""
+CLOSECONNECTIONS=""
+COMPRESSION=""
+
+while getopts "s:h:p:u:r:c:z:d:" OPTION; do
+    case $OPTION in
+    s  )
+        BACKUP_LOCATION="$OPTARG"
+        ;;
+    h  )
+        HOST="$OPTARG"
+        ;;
+    p  )
+        PORT="$OPTARG"
+        ;;
+    u  )
+        USERNAME="$OPTARG"
+        ;;
+    r  )
+        RANGERGROUP="$OPTARG"
+        ;;
+    c  )
+        CLOSECONNECTIONS="$OPTARG"
+        ;;
+    z  )
+        COMPRESSION="-Z $OPTARG"
+        ;;
+    \? ) echo "Unknown option: -$OPTNAME" >&2;
+         usage
+         exit 1;;
+    :  ) echo "Missing option argument for -$OPTARG" >&2; exit 1;;
+    esac
+done
+
+shift $((OPTIND-1))
+DATABASENAMES="$@"
+
+
+[ -z "$BACKUP_LOCATION" ]  || [[ $BACKUP_LOCATION == \-* ]] && doLog "backup location is not specified, use the -s option"
+[ -z "$RANGERGROUP" ] || [[ $RANGERGROUP == \-* ]] && doLog "Ranger admin group is not specified, use the -r option"
+[ -z "$CLOSECONNECTIONS" ] || [[ $CLOSECONNECTIONS == \-* ]] && doLog "Whether to close connections flag is not specified, use the -c option"
+
+[ -z "$BACKUP_LOCATION" ] || [ -z "$RANGERGROUP" ] || [ -z "$CLOSECONNECTIONS" ] && doLog "At least one mandatory parameter is not set!" && usage && exit 1
+
+doLog "Backup with compression level $COMPRESSION"
+
+# Root directory for local postgres dump.
+BACKUPS_DIR="/var/tmp/"
+# Directory for the current postgres dump.
+DATE_DIR=${BACKUPS_DIR}/$(date '+%Y-%m-%dT%H:%M:%SZ')
+
+{%- from 'postgresql/settings.sls' import postgresql with context %}
+{% if postgresql.ssl_enabled == True %}
+export PGSSLROOTCERT="{{ postgresql.root_certs_file }}"
+export PGSSLMODE=verify-full
+{%- endif %}
+
+errorExit() {
+  if [[ "$CLOSECONNECTIONS" == "true" ]]; then
+    limit_incomming_connection $SERVICE -1
+  fi
+  if [ -d "$DATE_DIR" ]; then
```

Comment:
Could we update this to `if [ -d "$DATE_DIR" ] && [[ $DATE_DIR == /var/tmp/* ]]` ? This was updated in [this](https://github.com/hortonworks/cloudbreak/pull/14528/files) PR and may have not been promoted to prod yet.

Reply:
done